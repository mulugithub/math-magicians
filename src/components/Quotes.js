import { useEffect, useState } from 'react';

const Quotes = () => {
  const baseURL = 'https://api.api-ninjas.com/v1/quotes';
  const [quotes, setQuote] = useState([]);
  const [error, setError] = useState('');
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    fetch(`${baseURL}?limit=2&?category=hope`, {
      headers: {
        'X-Api-Key': 'XFhfOEUxAb8ziROpQcmIXw==i6Vdbqw3WLtkON78',
      },
    })
      .then((result) => {
        setLoading(false);
        setError('');
        return result.json();
      })
      .then((data) => setQuote(data))
      .catch((err) => { setError(err); setLoading(false); });
  }, []);

  return (
    <div className="quotes">
      {error && <span className="error">Error while fetching data...</span>}
      {
        quotes.map((quote) => (
          <div className="quotes-wrapper" key={quote.quote}>
            <p className="quote">{quote.quote}</p>
            <small className="author">{quote.author}</small>
          </div>
        ))
        }
      {loading && <div>Loading...</div>}
    </div>
  );
};

export default Quotes;
