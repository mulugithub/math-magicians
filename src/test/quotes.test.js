import React from 'react';
import { render } from '@testing-library/react';
import Quotes from '../components/Quotes';

describe('Quotes component', () => {
  test('render Quotes correctly', () => {
    const { container } = render(<Quotes />);
    expect(container).toMatchSnapshot();
  });
});
