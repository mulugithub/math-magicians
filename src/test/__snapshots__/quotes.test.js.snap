// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes component render Quotes correctly 1`] = `
<div>
  <div
    class="quotes"
  >
    <div>
      Loading...
    </div>
  </div>
</div>
`;
